awx_collection/tools/integration_testing.yml

```diff
@@ -1,5 +1,28 @@
 ---
-- hosts: localhost
+# NOTE: Using a token makes tests run significantly faster
+- name: Generate an OAuth2 token for the test run
+  hosts: localhost
+  gather_facts: false
+  connection: local
+  collections:
+    - awx.awx
+  tasks:
+    - name: Create a new token using username/password
+      awx.awx.token:
+        description: 'Token used for collection tests {{ now() }}'
+        scope: write
+        state: present
+        controller_username: admin
+        controller_password: >-
+          {{ (lookup('ansible.builtin.file',
+          '{{ playbook_dir }}/../../tools/docker-compose/_sources/secrets/admin_password.yml') | from_yaml
```

Comment:
I'm a little cringe on this because I change my password with awx-manage but don't update the file which would affect me if I was running this locally. You could do something like:
```
controller_password: "{{ password_override | lookup() }}"
```